lib/irb/command/copy.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module IRB
+  module Command
+    class Copy < Base
+      category "Workspace"
+      description "Copy command output to clipboard"
+
+      help_message(<<~HELP)
+        Usage: copy (command)
+      HELP
+
+      def execute(arg)
+        value = irb_context.workspace.binding.eval(arg)
+        output = irb_context.inspect_method.inspect_value(value, colorize: false)
+
+        if clipboard_available?
+          copy_to_clipboard(output)
+          puts "Copied to system clipboard"
+        else
+          temp_file = write_to_tempfile(output)
+          puts "Wrote: #{temp_file}"
+        end
+      rescue StandardError => e
+        warn "Error: #{e}"
+      end
+
+      private
+
+      def copy_to_clipboard(text)
+        IO.popen(clipboard_program, 'w') do |io|
+          io.write(text)
+        end
+
+        raise IOError.new("Copying to clipboard failed") unless $? == 0
+      end
+
+      def write_to_tempfile(text)
+        file = Tempfile.new
+        file.write(text)
+        file.close
+        file.path
+      end
+
+      def clipboard_program
+        return "pbcopy" if executable?("pbcopy")
+        return "xclip -selection clipboard" if executable?("xclip")
```

Comment:
A potential future improvement: having a `IRB.conf` or env var to allow users specify the clipboard command to use.